Neos.Neos/Classes/Controller/Backend/ContentController.php

```diff
@@ -137,29 +132,21 @@ public function uploadAssetAction(Asset $asset, $metadata)
     {
         $this->response->setHeader('Content-Type', 'application/json');
 
-        /** @var Site $currentSite */
-        $currentSite = $this->siteRepository->findOneByNodeName($this->request->getInternalArgument('__siteNodeName'));
-        if ($currentSite !== null && $currentSite->getAssetCollection() !== null) {
-            $currentSite->getAssetCollection()->addAsset($asset);
-            $this->assetCollectionRepository->update($currentSite->getAssetCollection());
-        }
-
-        switch ($metadata) {
-            case 'Asset':
-                $result = $this->getAssetProperties($asset);
-                if ($this->persistenceManager->isNewObject($asset)) {
-                    $this->assetRepository->add($asset);
-                }
-                break;
-            case 'Image':
+        if ($metadata !== 'Asset' && $metadata !== 'Image') {
+            $this->response->setStatus(400);
+            $result = ['error' => 'Invalid "metadata" type: ' . $metadata];
+        } else {
+            if ($asset instanceof ImageInterface && $metadata === 'Image') {
                 $result = $this->getImageInterfacePreviewData($asset);
-                if ($this->persistenceManager->isNewObject($asset)) {
-                    $this->imageRepository->add($asset);
-                }
-                break;
-            default:
-                $this->response->setStatus(400);
-                $result = array('error' => 'Invalid "metadata" type: ' . $metadata);
+            } else {
+                $result = $this->getAssetProperties($asset);
+            }
+            $nodePath = $this->request->getInternalArgument('__nodePath');
+            $node = $this->propertyMapper->convert($nodePath, NodeInterface::class);
```

Comment:
any reason not to use normal property mapping for the $node?

Reply:
@aertmann What do you mean by "normal property mapping", having `node` as non-internal argument of the action? Good question, I think that could be changed.
Do you agree with the change apart from this?

Reply:
exactly, and yes looks good otherwise